reputation/reputation.go

```diff
@@ -201,16 +210,114 @@ func ModifyRep(ctx context.Context, conf *models.ReputationConfig, guildID int64
 	return
 }
 
-func insertUpdateUserRep(ctx context.Context, guildID, userID int64, amount int64) (err error) {
+const (
+	MaxRepRoles        = 5
+	MaxRepRolesPremium = 25
+)
+
+func GuildMaxRepRoles(guildID int64) int {
+	if isPrem, _ := premium.IsGuildPremium(guildID); isPrem {
+		return MaxRepRolesPremium
+	}
+	return MaxRepRoles
+}
+
+func UpdateRepRoles(gs *dstate.GuildSet, ms *dstate.MemberState, rep int64) error {
```

Comment:
This code is a bit nastier than one might imagine because it needs to handle the case where the target member is higher in the role hierarchy than the bot, in which case we cannot bulk edit the member's roles. I don't think there's a way to avoid this case; such a setup is not uncommon: we would, for instance, run into this case if we tried to set up rep roles on the support server.